framework/db/ActiveRecord.php

```diff
@@ -455,18 +455,28 @@ protected function insertInternal($attributes = null)
             }
         }
         $db = static::getDb();
-        $command = $db->createCommand()->insert($this->tableName(), $values);
-        if (!$command->execute()) {
+        // depending on the underlying DBMS, the command may return results
+        // or lastInsertId has to be used to fetch primary key values
+        $command = $db->createCommand()->insertReturning($this->tableName(), $values, $this->primaryKey());
+        // force preparing as a command for writing, not reading
+        $command->prepare(false);
```

Comment:
Is this line really needed? `$forRead` defaults to null, which means the command will check the SQL to determine which it is a read SQL.